core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2755,9 +2763,13 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
                                 petKey,
                                 expectedGoogleProjectPermissions
         )
-        .recoverWith { case t: IOException =>
-          // Throw a 400 to avoid Sentry notifications.
-          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, t)))
+        .recoverWith {
+          case t: GoogleJsonResponseException =>
+            val code = getStatusCodeHandlingUnknown(t.getStatusCode)
+            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(code, t.getDetails.toString)))
+          case t: IOException =>
```

Comment:
`GoogleJsonResponseException` extends `IOException`, so handle the more specific case first (we can get the status code for that!).